src/core/src/core_logic/RebootManager.py

```diff
@@ -14,116 +14,167 @@
 #
 # Requires Python 2.7+
 
-"""Reboot management"""
+""" Reboot management """
 import time
+
 from core.src.bootstrap.Constants import Constants
 
 
 class RebootManager(object):
-    """Implements the reboot management logic"""
-    def __init__(self, env_layer, execution_config, composite_logger, status_handler, package_manager, default_reboot_setting='IfRequired'):
+    """ Implements the reboot management logic """
+    def __init__(self, env_layer, execution_config, composite_logger, status_handler, package_manager, default_reboot_setting=Constants.REBOOT_IF_REQUIRED):
         self.execution_config = execution_config
 
         self.composite_logger = composite_logger
         self.package_manager = package_manager
         self.status_handler = status_handler
         self.env_layer = env_layer
 
-        self.minutes_to_shutdown = str((Constants.REBOOT_BUFFER_IN_MINUTES - 5) if (Constants.REBOOT_BUFFER_IN_MINUTES > 5) else Constants.REBOOT_BUFFER_IN_MINUTES)  # give at least 5 minutes for a reboot unless the buffer is configured to be lower than that
-        self.reboot_cmd = 'sudo shutdown -r '
-        self.maintenance_window_exceeded_flag = False
+        self.__reboot_cmd = 'sudo shutdown -r '
+        self.__maintenance_window_exceeded_flag = False   # flag to indicate if the maintenance window was exceeded **separately** at reboot manager
 
-        self.reboot_setting = self.sanitize_reboot_setting(self.execution_config.reboot_setting, default_reboot_setting)
+        self.__reboot_setting = self.sanitize_reboot_setting(self.execution_config.reboot_setting, default_reboot_setting)
 
+    # region - Reboot condition reporters
     @staticmethod
     def is_reboot_time_available(current_time_available):
+        # type: (int) -> bool
         """ Check if time still available for system reboot """
         return current_time_available >= Constants.REBOOT_BUFFER_IN_MINUTES
 
-    # REBOOT SETTING
-    # ==============
+    def is_reboot_pending(self):
+        # type: () -> bool
+        """ Check if a reboot is pending either from the package manager or the status handler """
+        return self.package_manager.force_reboot or (self.status_handler and self.status_handler.is_reboot_pending)
+
+    def has_maintenance_window_exceeded_at_reboot_manager(self):
+        # type: () -> bool
+        """ Check if the maintenance window was exceeded at reboot manager """
+        return self.__maintenance_window_exceeded_flag
+
+    def get_reboot_setting_sanitized(self):
```

Comment:
Why is this named get_reboot_setting_**sanitized** if it doesn't process/sanitize/format self.__reboot_setting ?

Reply:
The variable itself is set sanitized and protected. The function name here is to reflect that status outward.
![image](https://github.com/user-attachments/assets/89f7f05b-f1b4-4544-b608-671351f56a3d)

The ideal thing to do would be to name the variable to match that.